### PR TITLE
openposix: Add strlen() conformance tests and assertion.xml

### PR DESCRIPTION
**Pull request by padmashree9110**
Adds
	1> 1-1 test
	2> assertions.xml
	3> coverage.txt

Change-Id: I382bb7f9dc4d9c8d1c7622bf4a126a374a772560

### CODE REVIEW

File: testcases/open_posix_testsuite/conformance/interfaces/strlen/1-1.c [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+#define _XOPEN_SOURCE 700
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include "posixtest.h"
+
+#include "../testfrmw/testfrmw.h"
+#include "../testfrmw/testfrmw.c"
+
```
Comment (metan-ucw) [inline testcases/open_posix_testsuite/conformance/interfaces/strlen/1-1.c]: The testfrmw is deprecated, new test should not use it.

File: testcases/open_posix_testsuite/conformance/interfaces/strlen/1-1.c [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+{
+	int ret = 0, i = 0, actual_len = 0;
+	char sample_str_1[MAX_ARRAY_SIZE] = "HELLO ";
+	char sample_str_2[] = "WORLD";
+	char *sample_str_3 = NULL;
+
+	//Just to avoid compliler optimization, we are using strcat to get required string,
+	//instead of taking string directly.
+	sample_str_3 = strcat(sample_str_1, sample_str_2);
+
```
Comment (metan-ucw) [inline testcases/open_posix_testsuite/conformance/interfaces/strlen/1-1.c]: This is a bit ugly.

Also this really lacks reasonable coverage for the strlen(). Why do we try to assert a single pre-defined string when we can easily generate random strings from zero length even up to several kilobytes in a loop and assert that what we got from strlen is exactly the length of the string we generated.

File: testcases/open_posix_testsuite/conformance/interfaces/strlen/1-1.c [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+	ret = strlen(sample_str_3);
+
+	for (i = 0; sample_str_3[i]; i++) {
+		actual_len++;
+	}
```
Comment (metan-ucw) [inline testcases/open_posix_testsuite/conformance/interfaces/strlen/1-1.c]: We know the length of the string from a start, there is no need to reimplement naive strlen() here.

File: testcases/open_posix_testsuite/conformance/interfaces/strlen/1-1.c [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+	srand(time(NULL) * len);
+	output_string = malloc(sizeof(char) * (len + 1));
+	for (int i = 0; i < len; i++) {
+		key = rand() % string_len;
+		output_string[i] = input_string[key];
+	}
+	output_string[len] = '\0';
+	return output_string;
+}
+
```
Comment (metan-ucw) [inline testcases/open_posix_testsuite/conformance/interfaces/strlen/1-1.c]: This has the same problems as the strcpy() test.

**Comment by anitha-suresh** [comment]
@metan-ucw : Could you please review/merge the patch.

**Comment by padmashree9110** [comment]
@ngie-eign Can you please review this.

File: testcases/open_posix_testsuite/conformance/interfaces/strlen/1-1.c [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+	int i;
+	size_t obtained_len;
+
+	for (i = 0; i < STRING_MAX_LEN; i += STEP_COUNT) {
+		ret_str = random_string(i);
+		obtained_len = strlen(ret_str);
+
+		if (obtained_len != i) {
+			printf(TNAME " Test Failed, return value is not as expected, "
+						"for the given string. Expected string length: %d,"
```
Comment (ngie-eign) [inline testcases/open_posix_testsuite/conformance/interfaces/strlen/1-1.c]: Please reindent this line per LKML.

File: testcases/open_posix_testsuite/conformance/interfaces/strlen/1-1.c [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+						" But got: %ld.\n", i, obtained_len);
```
Comment (ngie-eign) [inline testcases/open_posix_testsuite/conformance/interfaces/strlen/1-1.c]: This should be `%zu`, not `%ld`, per printf(3).

File: testcases/open_posix_testsuite/conformance/interfaces/strlen/1-1.c [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+			exit(PTS_FAIL);
+		}
+		free(ret_str);
+	}
+	printf(TNAME " Test Passed, strlen() return value is as expected, "
+						"for the given strings.\n");
```
Comment (ngie-eign) [inline testcases/open_posix_testsuite/conformance/interfaces/strlen/1-1.c]: Please reindent per LKML.

**Comment by ngie-eign** [comment]
Please also add a `Signed-off-by:` line in the commit message, e.g., use `git commit --amend -s`.

**Comment by padmashree9110** [comment]
@ngie-eign I have updated the patch. can you please merge this patch.

**Pull request merged**